### PR TITLE
Fix psa_generate_key on Curve25519

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Fix https://github.com/ARMmbed/mbedtls/issues/3727

### CODE REVIEW

File: tests/suites/test_suite_psa_crypto.data [hunk 1]
```diff
@@ -2710,6 +2718,10 @@ depends_on:MBEDTLS_ECP_C:MBEDTLS_ECP_DP_SECP256R1_ENABLED:MBEDTLS_ECDSA_C
 # doesn't fully relate the curve with its size.
 generate_key:PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1):128:PSA_KEY_USAGE_EXPORT | PSA_KEY_USAGE_SIGN_HASH | PSA_KEY_USAGE_VERIFY_HASH:PSA_ALG_ECDSA_ANY:PSA_ERROR_NOT_SUPPORTED
 
+PSA generate key: ECC, Curve25519, good
+depends_on:MBEDTLS_ECP_C:MBEDTLS_ECP_DP_CURVE25519_ENABLED:MBEDTLS_ECDH_C
+generate_key:PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_MONTGOMERY):255:PSA_KEY_USAGE_EXPORT | PSA_KEY_USAGE_DERIVE:PSA_ALG_ECDH:PSA_SUCCESS
+
```
Comment (stevew817) [inline tests/suites/test_suite_psa_crypto.data:2724]: Maybe a good time to introduce the same for 448?

**Review by stevew817 — APPROVED** [review]

File: tests/suites/test_suite_psa_crypto.data (hunk 1 above)
Comment (gilles-peskine-arm) [reply tests/suites/test_suite_psa_crypto.data:2724]: Yes, but no: `mbedtls_ecp_read_key` doesn't support Curve448, and I don't know what else might be missing for Curve448. I didn't want to turn this into “add support for Curve448 everywhere it's missing”.

**Review by ronald-cron-arm — APPROVED** [review]
LGTM
I've checked the public, private, shared key against RFC 7748 for the two key agreement tests.
I've checked that on commit HEAD~2, the key generation test fails with psa_generate_key error code being NOT_SUPPORTED.
I've checked that on commit HEAD~1, the key generation test fails with psa_generate_key error code being INVALID_ARGUMENT and finally that the test pass on HEAD.
Otherwise, the arguments for the code changes make sense to me.

**Review by paul-elliott-arm — APPROVED** [review]
Changes make sense, and as above the tests pass.

**Comment by ronald-cron-arm** [comment]
All CI tests run successfully even if the status of continuous-integration/jenkins/pr-merge didn't pass through, thus ready to be merged.

**Pull request merged**